pkg/settlement/swap/swap.go

```
@@ -212,3 +223,73 @@ func (s *Service) Handshake(peer swarm.Address, beneficiary common.Address) erro
 	}
 	return nil
 }
+
+// LastSentCheque returns the last sent cheque for the peer
+func (s *Service) LastSentCheque(peer swarm.Address) (*chequebook.SignedCheque, error) {
+
+	common, known, err := s.addressbook.Beneficiary(peer)
+
+	if err != nil {
+		return nil, errors.New("error translating swarm address to chequebook address")
+	}
+
+	if !known {
+		return nil, chequebook.ErrNoCheque
+	}
+
+	return s.chequebook.LastCheque(common)
+}
+
+// LastReceivedCheque returns the last received cheque for the peer
+func (s *Service) LastReceivedCheque(peer swarm.Address) (*chequebook.SignedCheque, error) {
+
+	common, known, err := s.addressbook.Chequebook(peer)
+
+	if err != nil {
+		return nil, errors.New("error translating swarm address to chequebook address")
```

Word error is redundant in the error message and the original error is not propagated.

---

changed to return nil, err